docs/source/reference/pysupported.rst

```diff
@@ -711,6 +711,49 @@ The use of reference counted types, e.g. strings, in sets is unsupported.
    made to the set will not be visible to the Python interpreter until
    the function returns.
 
+.. _feature-typed-set:
+
+Typed Set
+----------
+
+.. note::
+  ``numba.typed.Set`` is an experimental feature, if you encounter any bugs in
+  functionality or suffer from unexpectedly bad performance, please report
+  this, ideally by opening an issue on the Numba issue tracker.
+
+As of version 0.58.0, a new implementation of the set data type is available,
+the *typed-set*. This is compiled library backed, type-homogeneous
+set data type that functions exactly like other typed data structures.
+You will need to import it explicitly from the `numba.typed` module::
+
+    In [1]: from numba.typed import Set
+
+    In [2]: from numba import njit, types
+
+    In [3]: @njit
+    ...: def foo(s):
+    ...:     s.add(23)
+    ...:     return s
+    ...:
+
+    In [4]: my_set = Set(types.intp)
```

Comment:
Question: can the `Set` also be initialized w/o a type and then be specialized with the first `add`?

Reply:
```
In [19]: from numba.typed import List

In [20]: l = List()

In [21]: l.append(1)

In [22]: l
Out[22]: ListType[int64]([1])
```